### DISCUSSION TITLE
AV detection and evasion

### DISCUSSION THREAD

**Discussion by DLK2** [UnamSanctam/UnamWebPanel#359] [General]
I've been testing this and the SilentMiner projects a lot lately and I gotta say I really admire your work. The miner lives up to the name and with some simple packing/crypting to evade static detections it works very well and manages to execute in the background of my VM without any interruptions or alerts by Windows Defender.
It's only a shame that if the goal is maximizing evasion then this neat web panel must be avoided. Even after adding an exclusion to the entire C drive and waiting for a few minutes it still gets detected in memory when the API Endpoint Url is enabled (currently the only option that gives away the miner fortunately). Maybe as a suggestion focusing on making this option somewhat more evasive would make this project a lot more practical and useful. Regardless, I admire your work and dedication. Best of luck!
![Detection](https://github.com/UnamSanctam/UnamWebPanel/assets/77547356/36d16a5b-73d2-4fc4-b6af-1724f70a661b)

**Comment by UnamSanctam** [comment]
Oh, you're getting detections from the http connections? I assume so since it only happens for you with the "API Endpoint", does it also happen with a "Remote Configuration"? Since they both are technically the same.

**Reply by DLK2** [reply]
not really. I tested both leaving only Remote Configuration and API Endpoint active but it got detected only with the latter. Does it maintain a connection to pastebin? Maybe it's just the connection duration to an unfamiliar domain that sets it off

**Reply by UnamSanctam** [reply]
Hmm, neither really maintains a connection, the web panel connection is just more frequent (once per minute). Does your web panel have https? Since otherwise the unencrypted traffic might be the reason.

**Reply by DLK2** [reply]
yes it was over HTTPS

**Reply by Extremal99** [reply]
any fix? @UnamSanctam  @DLK2

**Reply by alcinzal** [reply]
It might be because your domain is flagged and exists in a database. You can see if you find your domain in one of these places:
https://urlhaus.abuse.ch/browse/
https://threatfox.abuse.ch/browse/
https://threatfox.abuse.ch/browse/tag/UNAM/

You can also try to find your domain on Github, as some repositories store flagged domains. Simply search for your domain and select "Code" on the left side. If you get too many results, you can search for the domain specifically at organizations that deals with adblocking or link flagging, like these:
https://github.com/uBlockOrigin
https://github.com/AdguardTeam

**Reply by Extremal99** [reply]
So thats cus of the domain ?

**Reply by alcinzal** [reply]
> So thats cus of the domain ?

Maybe, I'm not quite sure, but that's my theory. Since I did not experience getting this notification from Windows Defender until after my domain got flagged.

**Reply by 0xSteff** [reply]
> Хм, ни один из них не поддерживает соединение, просто подключение к веб-панели происходит чаще (раз в минуту). У вашей веб-панели есть https? В противном случае причиной может быть незашифрованный трафик.

How often does connection to remote configuration occur?

**Reply by alcinzal** [reply]
> How often does connection to remote configuration occur?

The miner sends information to the web panel every 60 seconds.

**Reply by 0xSteff** [reply]
> > Как часто происходит подключение к удаленной конфигурации?
> 
> Майнер отправляет информацию на веб-панель каждые 60 секунд.

I need update time "Remote configuraton" not web panel

**Reply by alcinzal** [reply]
> I need update time "Remote configuraton" not web panel

Oh my bad misunderstood the question. I do think that is also once every 60 seconds, but might be wrong.

**Reply by 0xSteff** [reply]
> > Мне нужно обновить время "Удаленная настройка", а не веб-панель
> 
> Ой, я неправильно понял вопрос. Я тоже думаю, что это происходит каждые 60 секунд, но могу ошибаться.

If it were so, I wouldn’t ask) I look through the logger at the time of clicking on the link. Sometimes the interval is an hour and a half, sometimes about 3 or sometimes 2 hours

**Reply by 0xSteff** [reply]
4 hours have passed since the last request to the configuration and the worker still has not followed the link.

**Reply by UnamSanctam** [reply]
>How often does connection to remote configuration occur?

The "Remote Configuration" gets contacted every 100 minutes (to prevent bans).

>If it were so, I wouldn’t ask) I look through the logger at the time of clicking on the link. Sometimes the interval is an hour and a half, sometimes about 3 or sometimes 2 hours

If that's the case then it's unable to contact whatever website you use, the miner contacts the "Remote Configuration" every 100 minutes and the web panel every 60 seconds no matter what since they run in a separate thread.
